iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBDataRegionAsyncConnector.java

```diff
@@ -539,9 +539,20 @@ public synchronized void close() {
       tabletBatchBuilder.close();
     }
 
+    decreaseOrRemoveAsyncClient();
+
     super.close();
   }
 
+  private void decreaseOrRemoveAsyncClient() {
+    clientManager.decreaseAttributesCount();
+
+    if (clientManager.getAttributesCount() == 0) {
+      clientManager.removeAttributesCount();
+      clientManager.closeAndRemoveClientManager();
+    }
```

Comment:
Also, it is better to use the `IoTDBDataRegionAsyncConnector.class` lock instead of the `IoTDBDataRegionAsyncConnector` instance lock. The put and remove operations and the corresponding reference counts should be mutually exclusive.

Reply:
@luoluoyuyu It adds the synchronized keyword to the entire close method

Reply:
CreateClient and removeClient do not use the same lock, so it may cause a thread to successfully update CreateClient but fail to update the reference, causing the Client to be deleted when removing, causing NPE